app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb

```diff
@@ -149,8 +149,9 @@ def find_host_by_provider_id(provider_id)
     def scheme_to_provider_mapping
       @scheme_to_provider_mapping ||= begin
         {
-          'gce' => ['ManageIQ::Providers::Google::CloudManager'.safe_constantize, :name],
-          'aws' => ['ManageIQ::Providers::Amazon::CloudManager'.safe_constantize, :uid_ems],
+          'gce'   => ['ManageIQ::Providers::Google::CloudManager'.safe_constantize, :name],
+          'aws'   => ['ManageIQ::Providers::Amazon::CloudManager'.safe_constantize, :uid_ems],
+          'azure' => ['ManageIQ::Providers::Azure::CloudManager'.safe_constantize, :uid_ems],
```

Comment:
I think you want to refer to :ems_ref instead of :uid_ems. ems_ref is the attribute containing remote uuid of the provider entity, across all entities. 

:uid_ems can mean something else, e.g. it has tenant_id for Azure::CloudManager